### DISCUSSION TITLE
MET_build_components installation issue

### DISCUSSION THREAD

**Discussion by vimal295sen** [dtcenter/METplus#2330] [Installation]
Hi,

I am trying to install METplus component so I get the error and I have follow the following steps -:
1. Download METplus-11.1 
2. extract
3. cd Build_Component
4. change in env_vars.bash file content are following 

/soft/metplus-2/METplus-5.1.0/build_components> cat env_vars.bash
###### Environment examples for a non-module based system #####
export TEST_BASE=./
export COMPILER=gnu
export MET_SUBDIR=${TEST_BASE}
export MET_TARBALL=met.tar.gz
#export MET_PYTHON_CC=CC_FLAGS
#export MET_PYTHON_LD=LD_FLAGS
export MET_PYTHON_CC=-I/home/apps/SiteSoftwares/gnu/PYTHON/3.9.1/include/python3.9  -Wno-unused-result -Wsign-compare  -DNDEBUG -g -fwrapv -O3 -Wall
export MET_PYTHON_LD=-L/home/apps/SiteSoftwares/gnu/PYTHON/3.9.1/lib/\ -lpython3.9\  -lcrypt -lpthread\ -ldl\  -lutil\ -lm\ -lm -Xlinker
#### Python version on your system
#export USE_MODULES=TRUE
export PYTHON_MODULE=gnu_python_3.9.1
### Local system ${PYTHON_MODULE} install location for python libraries
#export MET_PYTHON=/usr/local/python3.6
#export PYTHON_MODULE_USE=1
#export SET_D64BIT==FALSE

5. download "compile_MET_all.sh.tgz" put into it into same location
6. after run the ./build_MET.sh
so IT's run initially but after some time stop with error message I have also attached all steps with error -:

 build_components]$ ./build_MET.sh
Processing externals description file : Externals.cfg
Checking status of externals: met, metviewer, metplotpy, metcalcpy, metdataio,
Checking out externals: met,

Command 'git clone --quiet https://github.com/dtcenter/MET MET'
from directory /home/vimal/soft/metplus-2/METplus-5.1.0
has taken 300 seconds. It may be hanging.

The command will continue to run, but you may want to abort
manage_externals with ^C and investigate. A possible cause of hangs is
when svn or git require authentication to access a private
repository. On some systems, svn and git requests for authentication
information will not be displayed to the user. In this case, the program
will appear to hang. Ensure you can run svn and git manually and access
all repositories without entering your authentication information.

metviewer, metplotpy, metcalcpy, metdataio,

--2023-08-30 15:06:03--  https://dtcenter.org/sites/default/files/community-code/met/compile_scripts/tar_files.tgz
Resolving dtcenter.org (dtcenter.org)... 23.185.0.2, 2620:12a:8001::2, 2620:12a:8000::2
Connecting to dtcenter.org (dtcenter.org)|23.185.0.2|:443... connected.
WARNING: cannot verify dtcenter.org's certificate, issued by ‘/C=US/O=Let's Encrypt/CN=R3’:
  Issued certificate has expired.
HTTP request sent, awaiting response... 200 OK
Length: 87786439 (84M) [text/plain]
Saving to: ‘tar_files.tgz’

100%[==========================================================================================>] 87,786,439   105MB/s   in 0.8s

2023-08-30 15:06:10 (105 MB/s) - ‘tar_files.tgz’ saved [87786439/87786439]

--2023-08-30 15:06:10--  http://dtcenter.org/sites/default/files/community-code/gfdl/standalone_gfdl-vortextracker_v3.9a.tar.gz
Resolving dtcenter.org (dtcenter.org)... 23.185.0.2, 2620:12a:8001::2, 2620:12a:8000::2
Connecting to dtcenter.org (dtcenter.org)|23.185.0.2|:80... connected.
HTTP request sent, awaiting response... 301 Moved Permanently
Location: https://dtcenter.org/sites/default/files/community-code/gfdl/standalone_gfdl-vortextracker_v3.9a.tar.gz [following]
--2023-08-30 15:06:11--  https://dtcenter.org/sites/default/files/community-code/gfdl/standalone_gfdl-vortextracker_v3.9a.tar.gz
Connecting to dtcenter.org (dtcenter.org)|23.185.0.2|:443... connected.
WARNING: cannot verify dtcenter.org's certificate, issued by ‘/C=US/O=Let's Encrypt/CN=R3’:
  Issued certificate has expired.
HTTP request sent, awaiting response... 200 OK
Length: 5365139 (5.1M) [application/gzip]
Saving to: ‘standalone_gfdl-vortextracker_v3.9a.tar.gz’

100%[==========================================================================================>] 5,365,139   --.-K/s   in 0.1s

2023-08-30 15:06:11 (45.9 MB/s) - ‘standalone_gfdl-vortextracker_v3.9a.tar.gz’ saved [5365139/5365139]

Extracting File
compile_MET_all.sh
cp: cannot stat ‘../MET/scripts/installation/compile_MET_all.sh’: No such file or directory
cp: cannot stat ‘../MET/scripts/installation/config’: No such file or directory
._tar_files
tar_files/
tar_files/._jasper-1.900.1.zip
tar_files/jasper-1.900.1.zip
tar_files/libpng-1.6.37.tar.gz
tar_files/._hdf5-1.8.18.tar.gz
tar_files/hdf5-1.8.18.tar.gz
tar_files/._gsl-1.16.tar.gz
tar_files/gsl-1.16.tar.gz
tar_files/cairo-1.16.0.tar.xz
tar_files/._gsl-2.1.tar.gz
tar_files/gsl-2.1.tar.gz
tar_files/._HDF-EOS2.16v1.00.tar.Z
tar_files/HDF-EOS2.16v1.00.tar.Z
tar_files/._gsl-1.11.tar.gz
tar_files/gsl-1.11.tar.gz
tar_files/._netcdf-4.4.1.1.zip
tar_files/netcdf-4.4.1.1.zip
tar_files/._freetype-2.6.5.tar.gz
tar_files/freetype-2.6.5.tar.gz
tar_files/._zlib-1.2.11.tar.gz
tar_files/zlib-1.2.11.tar.gz
tar_files/._pixman-0.40.0.tar.gz
tar_files/pixman-0.40.0.tar.gz
tar_files/._HDF4.2r3.tar.gz
tar_files/HDF4.2r3.tar.gz
tar_files/._netcdf-cxx4-4.3.0.tar.gz
tar_files/netcdf-cxx4-4.3.0.tar.gz
tar_files/._g2clib-1.6.0.tar
tar_files/g2clib-1.6.0.tar
tar_files/._BUFRLIB_v11-3-0.tar
tar_files/BUFRLIB_v11-3-0.tar
tar: met: Cannot stat: No such file or directory
tar: Exiting with failure status due to previous errors
env_vars.bash: line 8: export: `-Wno-unused-result': not a valid identifier
env_vars.bash: line 8: export: `-Wsign-compare': not a valid identifier
env_vars.bash: line 8: export: `-DNDEBUG': not a valid identifier
env_vars.bash: line 8: export: `-g': not a valid identifier
env_vars.bash: line 8: export: `-fwrapv': not a valid identifier
env_vars.bash: line 8: export: `-O3': not a valid identifier
env_vars.bash: line 8: export: `-Wall': not a valid identifier
env_vars.bash: line 9: export: `-lcrypt': not a valid identifier
env_vars.bash: line 9: export: `-lpthread -ldl ': not a valid identifier
env_vars.bash: line 9: export: `-lutil -lm -lm': not a valid identifier
env_vars.bash: line 9: export: `-Xlinker': not a valid identifier
./build_MET.sh: line 39: cd: met: No such file or directory
./build_MET.sh: line 40: ./bootstrap: No such file or directory
bash: compile_MET_all.sh: No such file or directory
[vimal@utility07 build_components]$

**Comment by jprestop** [comment]
@vimal295sen Apologies for the delay in responding to this Discussion. @fisherhucar I believe you are the best person to help out this user with manage externals.  Could you please take a look and follow up?

**Reply by hankenstein2** [reply]
Hello, can you try giving the MET_BASE a full path? You have export TEST_BASE=./
I usually need to set TEST_BASE=/path/directory/

**Reply by hankenstein2** [reply]
Hi again @vimal295sen do you need any more assistance or can we mark this issue closed?

**Reply by vimal295sen** [reply]
Sorry for the delay , but I have resolve my issue so now you can close the issue

**Comment by vimal295sen** [comment]
Dear support team,

Thank you for your reply now installation done, so you can close the issue.

Regards,
Vimal sen

**Reply by hankenstein2** [reply]
Thank you Vimal, let us know if we can help with any other part of MET/METplus
